prestoadmin/server.py

```diff
@@ -82,11 +86,202 @@ def new_sysnode_processor(node_info_rows):
 CONNECTOR_INFO_SQL = 'select catalog_name from system.metadata.catalogs'
 _LOGGER = logging.getLogger(__name__)
 
+DOWNLOAD_DIRECTORY = main_dir
+DEFAULT_RPM_NAME = 'presto-server-rpm.rpm'
+
+
+def _find_or_download_most_recent_presto_rpm():
+    latest_rpm_url = 'https://repository.sonatype.org/service/local/artifact/maven' \
+                     '/content?r=central-proxy&g=com.facebook.presto' \
+                     '&a=presto-server-rpm&e=rpm&v=RELEASE'
+    return _find_or_download_rpm(latest_rpm_url)
+
+
+def _check_if_valid_version(rpm_version):
+    """
+    This function returns whether the input rpm_version is valid.
+
+    The version is considered valid if it follows the format of
+    Semantic Versioning. That means it should have the form
+    X.Y.Z for Major.Minor.Patch. Z may not be applicable for
+    all rpm versions, so X.Y is also valid. X.Y.Z must all be numbers.
+
+    The version may be optionally followed by t (for teradata)
+    or f (for facebook), but no other letters are acceptable.
+    """
+    return re.match('^[0-9]+(\.[0-9]+){1,2}[tf]?$', rpm_version)
```

Comment:
This isn't the right regex for 148t and later teradata versions.  148t looks like 0.148-t.x.y